src/Parsers/IAST.cpp

```diff
@@ -217,38 +218,44 @@ String IAST::getColumnNameWithoutAlias() const
 }
 
 
-void IAST::FormatSettings::writeIdentifier(const String & name) const
+void IAST::FormatSettings::writeIdentifier(const String & name, bool ambiguous) const
```

Comment:
Consider searching for this identifier in the [list of keywords](https://github.com/ClickHouse/ClickHouse/blob/129d1e387e1b436ac572ca5db1a50dd9ed79ddae/src/Parsers/CommonParsers.h#L588) to check if it's ambiguous or not

Reply:
I already updated the PR to check if the identifier is one of the keywords.
Some keywords (I believe) are commonly used as identifiers `date`, `day`, `min`, `max`, `MONTH`, etc. So they will be quoted by default.